test/testdata/lsp/code_actions/extract_class_func.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Extract method to module
+    'Hello'
+  end
+  
+  sig do
+    params(x: String)
+    .returns(String)
+  end
+  def name(x)
+    "#{Foo.greeting} #{x}"
+  end
+end
+
+module A
+  class B
+    extend T::Sig
+
+    sig {void}
+    def bar
+      m = Foo
+      Foo.greeting
+
+      m.greeting
+      print((Foo if true).greeting)
```

Comment:
why is there no need for an error assertion here? Doesn't `greeting` not always exist on the receiver, because it might be `nil` sometimes?

Reply:
My goal with that case here was to test that renaming works correctly if the receiver is an 
expression. There is no error here, because we're being smart about the condition being true every time I guess. However, if you replace `true` with an expression, you'll actually get an error

<img width="953" alt="Screen Shot 2022-03-14 at 21 36 34" src="https://user-images.githubusercontent.com/9540120/158265453-909a0e86-0ef0-4d3b-ad98-a29b81a61281.png">